src/bn_balances.erl

```diff
@@ -0,0 +1,195 @@
+-module(bn_balances).
+
+-include("bn_jsonrpc.hrl").
+
+-behavior(gen_server).
+-export([start_link/1, init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+% api
+-export([get_historic_balance/2]).
+% hooks
+-export([incremental_commit_hook/1, end_commit_hook/2]).
+
+-define(DB_FILE, "balances.db").
+-define(SERVER, ?MODULE).
+
+-record(state, {
+    dir :: file:filename_all(),
+    db :: rocksdb:db_handle(),
+    default :: rocksdb:cf_handle(),
+    entries :: rocksdb:cf_handle(),
+    entry_heights ::rocksdb:cf_handle()
+}).
+
+%%
+%% Blockchain follower
+%%
+
+start_link(Args) ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, Args, []).
+
+init(Args) ->
+    Dir = filename:join(proplists:get_value(base_dir, Args, "data"), ?DB_FILE),
+    case load_db(Dir) of
+        {ok, State} ->
+            persistent_term:put(?MODULE, State),
+            {ok, State};
+        Error ->
+            Error
+    end.
+
+handle_call(Request, _From, State) ->
+    lager:notice("Unhandled call ~p", [Request]),
+    {reply, ok, State}.
+
+handle_cast(Msg, State) ->
+    lager:notice("Unhandled cast ~p", [Msg]),
+    {noreply, State}.
+
+handle_info(Info, State) ->
+    lager:notice("Unhandled info ~p", [Info]),
+    {noreply, State}.
+
+terminate(_Reason, #state{db = DB}) ->
+    rocksdb:close(DB).
+
+%%
+%% Hooks
+%%
+
+incremental_commit_hook(Changes) -> 
+    {ok, #state{db=DB, default=DefaultCF, entries=EntriesCF, entry_heights=EntryHeightsCF}} = get_state(),
+    Chain = blockchain_worker:blockchain(),
+    {ok, Height} = blockchain:height(Chain),
```

Comment:
OK i see. I was getting some issues with the block follower before because I was using an async follower. I changed it to be a sync follower and will update.